pkg/cluster/spec/util.go

```diff
@@ -45,7 +43,49 @@ type ClusterMeta struct {
 	Topology *Specification `yaml:"topology"`
 }
 
+var _ UpgradableMetadata = &ClusterMeta{}
+
+// SetVersion implement UpgradableMetadata interface.
+func (m *ClusterMeta) SetVersion(s string) {
+	m.Version = s
+}
+
+// SetUser implement UpgradableMetadata interface.
+func (m *ClusterMeta) SetUser(s string) {
+	m.User = s
+}
+
+// GetTopology implement Metadata interface.
+func (m *ClusterMeta) GetTopology() Topology {
+	return m.Topology
+}
+
+// SetTopology implement Metadata interface.
+func (m *ClusterMeta) SetTopology(topo Topology) {
+	tidbTopo, ok := topo.(*Specification)
+	if !ok {
+		panic(fmt.Sprintln("wrong type: ", reflect.TypeOf(topo)))
+	}
+
+	m.Topology = tidbTopo
+}
+
+// GetBaseMeta implements Metadata interface.
+func (m *ClusterMeta) GetBaseMeta() *BaseMeta {
+	return &BaseMeta{
+		Version: m.Version,
+		User:    m.User,
+		OpsVer:  &m.OpsVer,
+	}
+}
+
+// AuditDir return the directory for saving audit log.
+func AuditDir() string {
+	return filepath.Join(profileDir, TiOpsAuditDir)
+}
+
 // SaveClusterMeta saves the cluster meta information to profile directory
+// TODO remove this
```

Comment:
It seems this is still referenced by other code. Why we need remove it?

Reply:
https://github.com/pingcap/tiup/pull/587/commits/17ce8e0c3c3953522ddc03d32fda98199da3a93d
It will better to remote all ClusterMeta into components/cluster
at least not utils.go